src/databricks/labs/ucx/mixins/dbfspath.py

```diff
@@ -0,0 +1,363 @@
+from __future__ import annotations
+
+import fnmatch
+import logging
+import os
+import posixpath
+import re
+
+from pathlib import Path, PurePath
+from typing import NoReturn
+
+from databricks.sdk import WorkspaceClient
+
+logger = logging.getLogger(__name__)
+
+
+class DbfsPath(Path):
```

Comment:
i stronlgy suggest extending the `WorkspacePath` into `DbfsPath` - there are 11 occurences of `self._ws.workspace` and they'll get transformed almost 1-1 to `self._ws.dbfs`